pkg/api/customization/monitor/project_graph_action.go

```diff
@@ -59,41 +60,65 @@ func (h *ProjectGraphHandler) QuerySeriesAction(actionName string, action *types
 		return err
 	}
 
-	prometheusName, prometheusNamespace := monitorutil.ClusterMonitoringInfo()
-	token, err := getAuthToken(userContext, prometheusName, prometheusNamespace)
-	if err != nil {
-		return err
-	}
-
 	reqContext, cancel := context.WithTimeout(context.Background(), prometheusReqTimeout)
 	defer cancel()
 
-	svcName, svcNamespace, svcPort := monitorutil.ClusterPrometheusEndpoint()
-	prometheusQuery, err := NewPrometheusQuery(reqContext, clusterName, token, svcNamespace, svcName, svcPort, h.dialerFactory, userContext)
-	if err != nil {
-		return err
-	}
+	var svcName, svcNamespace, svcPort, token string
+	var queries []*PrometheusQuery
+	if inputParser.Input.Filters["resourceType"] == "istioproject" {
+		if inputParser.Input.MetricParams["namespace"] == "" {
+			return fmt.Errorf("no namespace found")
+		}
 
-	var graphs []mgmtclientv3.ProjectMonitorGraph
-	err = access.List(apiContext, apiContext.Version, mgmtclientv3.ProjectMonitorGraphType, &types.QueryOptions{Conditions: inputParser.Conditions}, &graphs)
-	if err != nil {
-		return err
-	}
+		svcName, svcNamespace, svcPort = monitorutil.IstioPrometheusEndpoint()
 
-	mgmtClient := h.clustermanager.ScaledContext.Management
-	var queries []*PrometheusQuery
-	for _, graph := range graphs {
-		g := graph
-		_, projectName := ref.Parse(graph.ProjectID)
-		refName := getRefferenceGraphName(projectName, graph.Name)
-		monitorMetrics, err := graph2Metrics(userContext, mgmtClient, clusterName, g.ResourceType, refName, graph.MetricsSelector, graph.DetailsMetricsSelector, inputParser.Input.MetricParams, inputParser.Input.IsDetails)
+		mgmtClient := h.clustermanager.ScaledContext.Management
+		istioGraphs, err := mgmtClient.ClusterMonitorGraphs(clusterName).List(metav1.ListOptions{LabelSelector: "component=istio"})
+		if err != nil {
+			return fmt.Errorf("list istio graph failed, %v", err)
+		}
+		for _, graph := range istioGraphs.Items {
+			_, projectName := ref.Parse(inputParser.Input.Filters["projectId"])
```

Comment:
Can we use `inputParser.ProjectID`?

Reply:
Of couse, we can use `inputParser.ProjectID`, maybe it is better to use it. I will update it.

Reply:
Updated!